### ISSUE TITLE
[@typescript-eslint/no-var-requires] Error as function arg, but not as constructor arg

### ISSUE THREAD

**Issue by adam1658** [typescript-eslint/typescript-eslint#3883]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended"],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json"
  },
  "plugins": ["@typescript-eslint"],
  "rules": {}
}
```

```TS
const configValidator = new Validator(require('./a.json'));
configValidator.addSchema(require('./a.json'));
```

tsconfig.json:
```JSON
{
  "compilerOptions": {}
}
```

**Expected Result**

Two errors - one about each require statement on lines one and two

**Actual Result**

Just one error - for line two.
If I changed line two to be a constructor call instead, the error would go away.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.31.1` |
| `@typescript-eslint/parser`        | `4.31.1` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `12` |
